### PR TITLE
[export] Log module hierarchy.

### PR DESCRIPTION
**Pull request by zhxchen17**
Summary:
We can also log the module hierarchy in the following format:
```
:ToplevelModule
sparse:SparshArch
dense:DenseArch
```
So that we can have more information recorded about model's identity.

Test Plan: CI

Differential Revision: D54921097

### CODE REVIEW

File: torch/export/_trace.py [hunk 1]
```diff
@@ -312,6 +313,30 @@ def _restore_state_dict(
     traced_module.recompile()
 
 
+def _get_module_hierarchy(mod: torch.nn.Module) -> Dict[str, str]:
+    ret = {}
```
Comment (Skylion007) [inline torch/export/_trace.py:317]: nit can be a dict comprehension

**Review by angelayi — APPROVED** [review]

**Comment by izaitsevfb** [comment]
@angelayi , could you please add relevant label and re-merge?
